addons/hr_expense/models/hr_expense.py

```diff
@@ -161,7 +160,8 @@ def _onchange_product_id(self):
 
     @api.onchange('company_id')
     def _onchange_expense_company_id(self):
-        self.employee_id = self.env['hr.employee'].search([('user_id', '=', self.env.uid), ('company_id', '=', self.company_id.id)])
+        if not self._context.get("default_employee_id"):
```

Comment:
2 small cosmetic notes here:
  - `self._context` is from the old API, instead use `self.env.context`
  - prefer simple quotes as double quotes, to be more consistent with the rest